Resources/public/js/views/services/ez-dashboardblocksviewservice.js

```diff
@@ -20,5 +23,163 @@ YUI.add('ez-dashboardblocksviewservice', function (Y) {
      * @constructor
      * @extends eZ.ViewService
      */
-    Y.eZ.DashboardBlocksViewService = Y.Base.create('dashboardBlocksViewService', Y.eZ.ViewService, [], {});
+    Y.eZ.DashboardBlocksViewService = Y.Base.create('dashboardBlocksViewService', Y.eZ.ViewService, [], {
+        _load: function (next) {
+            this._loadRootLocation()
+                .then(Y.bind(this._loadRootLocationModelData, this))
+                .then(next)
+                .catch(Y.bind(this._handleError, this));
+        },
+
+        /**
+         * Loads content root location
+         *
+         * @method _loadRootLocation
+         * @protected
+         * @return {Y.Promise}
+         */
+        _loadRootLocation: function () {
+            var errorData = {
+                message: this.get('loadRootLocationErrorMessage'),
+                identifier: this.get('loadRootLocationErrorIdentifier')
+            };
+
+            return new Y.Promise(Y.bind(function (resolve, reject) {
+                this.get('capi').getDiscoveryService().getInfoObject('rootLocation', function (error, response) {
+                    if (error) {
+                        reject(errorData, error);
+
+                        return;
+                    }
+
+                    resolve(response._href);
+                });
+            }, this));
+        },
+
+        /**
+         * Loads model data of a root location
+         *
+         * @method _loadRootLocationModelData
+         * @protected
+         * @param rootLocationId {String} root location REST id
+         * @return {Y.Promise}
+         */
+        _loadRootLocationModelData: function (rootLocationId) {
+            var rootLocation = this.get('rootLocation'),
+                errorData = {
+                    message: this.get('loadLocationModelErrorMessage'),
+                    identifier: this.get('loadLocationModelErrorIdentifier')
+                };
+
+            rootLocation.set('id', rootLocationId);
+
+            return new Y.Promise(Y.bind(function (resolve, reject) {
+                rootLocation.load({api: this.get('capi')}, function (error, response) {
+                    if (error) {
+                        reject(errorData, error);
+
+                        return;
+                    }
+
+                    resolve(response.document.Location);
```

Comment:
the response parsing is the responsibility of the Location model. Writing `response.document.Location` is the very beginning of parsing the response so that should not appear here, I think you should write:
```js
resolve(rootLocation);
```

since the result of the Promise here is supposed to be a fully loaded root Location.